src/stories/3-ContactTop.stories.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import { withKnobs, boolean } from '@storybook/addon-knobs';
+import data from './contactTopData.json';
+import { ContactTop } from '../otherPagesComponent/ContactTop';
+
+export default {
+  title: 'ContactTop',
+  component: ContactTop,
+  decorators: [withKnobs]
+};
+
+export const CotactTopComponent = () => {
+  return (
+    <ContactTop
+      socialIconsList={data}
+      loading={boolean('loading', false)}
+      error={boolean('error', false)}
+      clicked={boolean('clicked', false)}
```

Comment:
when you want to test event handlers like this one `clicked`, you should use storybook's [actions addon](https://github.com/storybookjs/storybook/tree/master/addons/actions)

```js
import { action } from '@storybook/addon-actions';
<ContactTop
      socialIconsList={data}
      loading={boolean('loading', false)}
      error={boolean('error', false)}
      clicked={action('Yellow-button-clicked')} //like this
/>
```

once you click on the button from within the story, you should see something like this in the actions panel
![image](https://user-images.githubusercontent.com/1198975/81973921-23fd9880-9625-11ea-85fd-29c060ad031a.png)

and in from the component implementation point of view, you should keep everything as it is like this:

```js<button
        onClick={clicked}
        className="btn w-2/12 h-full text-c100 text-sm font-bold bg-c300"
      >
        Start Donation
      </button>```

Reply:
Thanks a lot i will  reconsider that
i would use action but i have confusing about the benefit of this .. i thought that I'm going to see the changes i wanted in the component ( like make section's background yellow )
then when i didn't get it .. i ignore using it
but now i know that this is just to tell me that this specific button has been clicked
Thanks again 👍 💯 
@Nilegfx

Reply:
think about it, if you want to make the background to be changed upon clicking the button, do you need to send a function from the outside into the component?

The reason why we send functions from outside is to invoke it to do things in a higher/another scope, the reason why storybook actions add-on exists is to tell you that your function was called and it shows you with which parameters it was called.

**an example**

if you changed the implementation of the component to be like: 
```js
//...
const ContactTop = ({ socialIconsList, loading, error, clicked }) => {

  // notice 1
  const yellowButtonClickHandler = () => {
    clicked(5);
  };

//...

<button
onClick={yellowButtonClickHandler} //notice 2
className="btn w-2/12 h-full text-c100 text-sm font-bold bg-c300"
>
Start Donation
</button>
}
```

and changed the container implementation to be like: 
```js
const ContactTopContainer = () => {
  const { data, loading, dataError } = useCharityAPI('/socialmedias');
  const [yellowButtonClickCounter, setYellowButtonClickCounter] = useState(0); //notice 3

  //notice 4
  const incrementYellowButtonCounterBy = number => {
    setYellowButtonClickCounter(yellowButtonClickCounter + number);
  };

  return (
    <ContactTop
      socialIconsList={data}
      loading={loading}
      error={dataError}
      clicked={incrementYellowButtonCounterBy} //notice 5
    />
  );
};
```

now in the story, when you click the button, the action panel will show you:
![image](https://user-images.githubusercontent.com/1198975/81981899-3a115600-9631-11ea-99e5-86a6a3d59fc0.png)


This is the benefit of it, now you know that your component called the function which it received as prop, which eventually passes it back to the parent/container component with the correct desired parameter